torchci/lib/bot/utils.ts

```diff
@@ -270,7 +270,14 @@ export async function hasApprovedPullRuns(
   if (pr_runs == null || pr_runs?.length == 0) {
     return false;
   }
-  return pr_runs.every((run) => run.conclusion != "action_required");
+  return !(
+    pr_runs.some((run) => run.conclusion === "action_required") ||
+    // Everything being a start up failure could hide the need for workflow
+    // approval
+    pr_runs.every(
+      (run) => run.conclusion === "failure" && run.created_at == run.updated_at
```

Comment:
Why is the `run.created_at == run.updated_at` check needed?

Reply:
It's to differentiate between an actual failure and a workflow start up failure